subsys/fs/shell.c

```diff
@@ -334,6 +334,66 @@ static int cmd_read(const struct shell *shell, size_t argc, char **argv)
 	return 0;
 }
 
+static int cmd_cat(const struct shell *shell, size_t argc, char **argv)
+{
+	char path[MAX_PATH_LEN];
+	struct fs_dirent dirent;
+	struct fs_file_t file;
+	int count;
+	int err;
+
+	for (size_t i = 1; i < argc; ++i) {
+		create_abs_path(argv[i], path, sizeof(path));
+
+		err = fs_stat(path, &dirent);
+		if (err) {
+			shell_error(shell, "Failed to obtain file %s (err: %d)",
+					path, err);
+			continue;
+		}
+
+		if (dirent.type != FS_DIR_ENTRY_FILE) {
+			shell_error(shell, "Note a file %s", path);
+			continue;
+		}
+
+		fs_file_t_init(&file);
+		err = fs_open(&file, path, FS_O_READ);
+		if (err) {
+			shell_error(shell, "Failed to open %s (%d)", path, err);
+			continue;
+		}
+
+		count = INT_MAX;
+
+		while (count > 0) {
+			ssize_t read;
+			uint8_t buf[16];
+
+			read = fs_read(&file, buf, MIN(count, sizeof(buf)));
+			if (read == 0) {
+				break;
+			}
+
+			if (read < 0) {
+				shell_error(shell, "Failed to read from file %s (err: %d)",
+					path, read);
+				break;
+			}
+
+			for (int j = 0; j < read; j++) {
+				shell_fprintf(shell, SHELL_NORMAL, "%c", buf[j]);
+			}
```

Comment:
Printing each character requires invoking entire call and reinterpretation of the formatting string. Would you consider always terminating the `buf` with 0 and printing it with `"%s"` in single `shell_fprintf` call each time you are able to obtain full buffer?

Reply:
@de-nordic Thanks for the review! I think that there will be a problem if the file is binary. No? I also fixed the other issues

Reply:
Ah, I have forgotten about the binary files. That is true.